src/Image.cc

```diff
@@ -200,8 +200,10 @@ Image::clearData() {
     _surface = NULL;
   }
 
-  free(_data);
-  _data = NULL;
+  if (_data) {
+    delete[] _data;
+    _data = nullptr;
+  }
```

Comment:
Nice 🙌 

btw. I think it's safe to run `delete` on a `nullptr`, so I don't think you need the `if`? 🤔

Reply:
Yeah, it's defined behavior. Removed the guard.